CID: 1351700 : Result not floating point

aujjwal-redhat opened the pull request:
Description:
The result of the division is truncated to an integer (a whole number), which is usually a loss of precision in a calculation.
Type-casted bricks_to_use to double to prevent the loss of precision.

Updates: #1060

Change-Id: I7e6aa29dbe27f2bb8e17231cc87c405798557694
Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

aujjwal-redhat committed:
CID: 1351700

Description:
The result of the division is truncated to an integer (a whole number), which is usually a loss of precision in a calculation.
Type casted bricks_to_use to double.

Updates: #1060

Change-Id: I7e6aa29dbe27f2bb8e17231cc87c405798557694
Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

xhernandez commented on xlators/cluster/dht/src/dht-selfheal.c:
```diff
@@ -1792,7 +1792,7 @@ dht_selfheal_layout_new_directory(call_frame_t *frame, loc_t *loc,
     } else {
         weight_by_size = _gf_false;
-        chunk = ((unsigned long)0xffffffff) / bricks_to_use;
+        chunk = ((unsigned long)0xffffffff) / ((double)bricks_to_use);
```
```suggestion
        chunk = ((double)0xffffffff) / ((double)bricks_to_use);
```

aujjwal-redhat replied:
Thanks @xhernandez. Will make the changes. :)

aujjwal-redhat committed:
CID: 1351700 : Result not floating point

Description:
The result of the division is truncated to an integer (a whole number), which is usually a loss of precision in a calculation.

Fix:
Type-casted bricks_to_use to double and replaced unsigned long with
double to prevent the loss of precision.

Updates: #1060

Signed-off-by: Ayush Ujjwal aujjwal@redhat.com

Change-Id: I917269e565ab86c2b3ff714ee20469f37317488b
Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

xhernandez commented:
/run regression

gluster-ant commented:
2 test(s) needed retry 
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/basic/glusterd/thin-arbiter-volume-probe.t
